src/test/java/com/fasterxml/jackson/databind/ser/SimpleFilterProviderTest.java

```diff
@@ -0,0 +1,138 @@
+package com.fasterxml.jackson.databind.ser;
+
+import com.fasterxml.jackson.annotation.JsonFilter;
+import com.fasterxml.jackson.databind.BaseMapTest;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.fasterxml.jackson.databind.ser.impl.SimpleBeanPropertyFilter;
+import com.fasterxml.jackson.databind.ser.impl.SimpleFilterProvider;
+
+/**
+ * Tests {@link SimpleFilterProvider} on registration of filters.
+ */
+public class SimpleFilterProviderTest extends BaseMapTest
+{
+
+    private final ObjectMapper MAPPER = newJsonMapper();
+
+    @JsonFilter("filterB")
+    public static class AnyBeanB
+    {
+        public String a;
+        public String b;
+
+        public AnyBeanB(String a, String b) {
+            this.a = a;
+            this.b = b;
+        }
+    }
+
+    public void testAddFilterLastOneRemains() throws Exception {
+        FilterProvider prov = new SimpleFilterProvider()
+                .addFilter("filterB", SimpleBeanPropertyFilter.serializeAll())
+                .addFilter("filterB", SimpleBeanPropertyFilter.filterOutAllExcept());
+        AnyBeanB beanB = new AnyBeanB("1a", "2b");
+
+        String jsonString = MAPPER.writer(prov).writeValueAsString(beanB);
+
+        assertEquals(a2q("{}"), jsonString);
+    }
+
+    public void testAddFilterLastOneRemainsFlip() throws Exception {
+        FilterProvider prov = new SimpleFilterProvider()
+                .addFilter("filterB", SimpleBeanPropertyFilter.filterOutAllExcept("a"))
+                .addFilter("filterB", SimpleBeanPropertyFilter.serializeAll());
+        AnyBeanB beanB = new AnyBeanB("1a", "2b");
+
+        String jsonString = MAPPER.writer(prov).writeValueAsString(beanB);
+
+        assertEquals(a2q("{'a':'1a','b':'2b'}"), jsonString);
+    }
+
+    public void testAddFilterNulls() throws Exception {
+        FilterProvider prov = new SimpleFilterProvider()
+                .addFilter("filterB", SimpleBeanPropertyFilter.filterOutAllExcept(null, null));
+        AnyBeanB beanB = new AnyBeanB("1a", "2b");
+
+        String jsonString = MAPPER.writer(prov).writeValueAsString(beanB);
+
+        assertEquals(a2q("{}"), jsonString);
+    }
+
+    public void testAddFilterEmptyString() throws Exception {
+        FilterProvider prov = new SimpleFilterProvider()
+                .addFilter("filterB", SimpleBeanPropertyFilter.filterOutAllExcept("", ""));
+        AnyBeanB anyBeanB = new AnyBeanB("1a", "2b");
+
+        String jsonString = MAPPER.writer(prov).writeValueAsString(anyBeanB);
+
+        assertEquals(a2q("{}"), jsonString);
+    }
+
+    @JsonFilter(value = "")
```

Comment:
Oh. The usual convention with tests is for helper POJOs to be declared before test methods.
Could you move these similarly to the top of test class, instead of between methods?